src/main/java/com/amazon/opendistroforelasticsearch/security/auditlog/impl/AuditMessage.java

```diff
@@ -323,12 +327,25 @@ void addRestMethod(final RestRequest.Method method) {
 
     void addRestRequestInfo(final RestRequest request, final AuditConfig.Filter filter) {
         if (request != null) {
-            addPath(request.path());
+            final String path = request.path();
+            addPath(path);
             addRestHeaders(request.getHeaders(), filter.shouldExcludeSensitiveHeaders());
             addRestParams(request.params());
             addRestMethod(request.method());
             if (filter.shouldLogRequestBody() && request.hasContentOrSourceParam()) {
-                addTupleToRequestBody(request.contentOrSourceParam());
+                try {
+                    final Tuple<XContentType, BytesReference> xContentTuple = request.contentOrSourceParam();
+                    final String requestBody =  XContentHelper.convertToJson(xContentTuple.v2(), false, xContentTuple.v1());
+                    if (path != null && requestBody != null
+                            && SENSITIVE_PATHS.matcher(path).find()
+                            && requestBody.contains(SENSITIVE_KEY)) {
+                        auditInfo.put(REQUEST_BODY, SENSITIVE_REPLACEMENT_VALUE);
+                    } else {
+                        auditInfo.put(REQUEST_BODY, requestBody);
```

Comment:
In case requestBody is null, this will work right?

Reply:
Actually requestBody must not be null there as per original behavior. I can remove the check to have the same original behavior